src/naive/date.rs

```diff
@@ -50,6 +50,26 @@ const MIN_DAYS_FROM_YEAR_0: i32 = (MIN_YEAR + 400_000) * 365 + (MIN_YEAR + 400_0
 #[cfg(test)] // only used for testing, but duplicated in naive::datetime
 const MAX_BITS: usize = 44;
 
+/// A week represented by a [`NaiveDate`] and a [`Weekday`] which is the first
+/// day of the week.
+#[derive(Debug)]
+pub struct NaiveWeek {
+    date: NaiveDate,
+    weekday: Weekday,
+}
+
+impl Weeklike for NaiveWeek {
+    type Day = NaiveDate;
+
+    #[inline]
+    fn start_day(&self) -> Self::Day {
+        let start = self.date.weekday().num_days_from_monday() as i64;
+        let end = self.weekday.num_days_from_monday() as i64;
+        let days = start - end;
+        self.date - Duration::days(if days >= 0 { days } else { 7 + days })
```

Comment:
I think `if days >= 0 { days } else { 7 + days }` could be more naturally expressed with some kind of modulo?